### PR TITLE
File and Block Hourly billing feature changes for slci

### PR DESCRIPTION
**Pull request by svrajput**
Please merge after Davids pull request to see actual changes (  https://github.com/softlayer/softlayer-python/pull/838 )

### CODE REVIEW

**Comment by svrajput** [comment]
Yeah sure I will take care of conflicts and shall update pull request.

File: tests/managers/file_tests.py [hunk 1]
```diff
@@ -166,6 +181,27 @@ def test_cancel_snapshot_immediately(self):
...
+            'billingItem': {
+                'activeChildren': [
+                    {'categoryCode': 'storage_snapshot_space', 'id': 417}
+                ],
+                'hourlyFlag': True,
+                'id': 449
+            },
+        }
+
+        self.file.cancel_snapshot_space(1234, immediate=True)
```
Comment (svrajput) [inline tests/managers/file_tests.py]: For hourly billing we can cancel snapshot immediate, its part of requirement.

File: tests/managers/block_tests.py [hunk 2]
```diff
@@ -181,6 +196,27 @@ def test_cancel_snapshot_immediately(self):
...
+            'billingItem': {
+                'activeChildren': [
+                    {'categoryCode': 'storage_snapshot_space', 'id': 417}
+                ],
+                'hourlyFlag': True,
+                'id': 449
+            },
+        }
+
+        self.block.cancel_snapshot_space(1234, immediate=True)
```
Comment (svrajput) [inline tests/managers/block_tests.py]: Its correct, user can cancel hourly billing instance on immediate basis. its part of requirement.

File: SoftLayer/CLI/block/order.py [hunk 3]
```diff
@@ -56,13 +56,28 @@
                   'storage_as_a_service',
                   'enterprise',
                   'performance']))
+@click.option('--billing',
+              type=click.Choice(['hourly', 'monthly']),
+              default='monthly',
+              show_default=True,
+              help="Optional parameter for Billing rate. Default to monthly")
```
Comment (svrajput) [inline SoftLayer/CLI/block/order.py]: show_default=True - removed as per discussion with David. will push latest file in a while.

File: tests/managers/block_tests.py (hunk 2 above)
Comment (dpickle2) [inline tests/managers/block_tests.py]: Based on the current logic in the Block (and File) managers, immediate will always be used if the volume has hourly billing. Passing in False in the unit test is meant to ensure that the immediate flag is correctly changed to be True when the API call is actually made.

**Comment by svrajput** [comment]
Unstable build .. hence closing. will create new pull request.

**Pull request closed without merging**